go/vt/vttablet/tabletserver/query_executor.go

```diff
@@ -748,7 +749,11 @@ type poolConn interface {
 
 func (qre *QueryExecutor) execSQL(conn poolConn, sql string, wantfields bool) (*sqltypes.Result, error) {
 	defer qre.logStats.AddRewrittenSQL(sql, time.Now())
-	return conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	res, err := conn.Exec(qre.ctx, sql, int(qre.tsv.qe.maxResultSize.Get()), wantfields)
+	if res != nil && int64(len(res.Rows)) > qre.tsv.qe.warnResultSize.Get() {
+		log.Warningf("Results returned exceeds warning threshold: %q", sql)
```

Comment:
Alternatively maybe you could do `log.SetOutput` at the start of your test to intercept the log stream with your own writer. Or use the `glog.Stats.WARNING.Lines()` function to see that a warning is produced if not check the actual string.